pkgdb/src/lock-flake-installable.cc

```diff
@@ -0,0 +1,439 @@
+          return outputNamesSet;
+        } },
+      outputSpec.raw() );
+  }
+
+  std::string systemAttribute;
+  {
+    auto systemCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "system" ) );
+
+    if ( ! systemCursor )
+      {
+        throw LockFlakeInstallableException(
+          nix::fmt( "could not find '%s.%s' in derivation",
+                    lockedAttrPath,
+                    "system" ) );
+      }
+    systemAttribute = ( *systemCursor )->getString();
+  }
+
+  // Read `name` field - field is impliend by the derivation
+  std::string name;
+  {
+    auto nameCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "name" ) );
+
+    if ( ! nameCursor )
+      {
+        throw LockFlakeInstallableException(
+          nix::fmt( "could not find '%s.%s' in derivation",
+                    lockedAttrPath,
+                    "name" ) );
+      }
+    name = ( *nameCursor )->getString();
+  }
+
+  // Read `pname` field
+  std::optional<std::string> pname;
+  {
+    auto pnameCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "pname" ) );
+
+    if ( pnameCursor ) { pname = ( *pnameCursor )->getString(); }
+  }
+
+  // Read `version` field
+  std::optional<std::string> version;
+  {
+    auto versionCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "version" ) );
+
+    if ( versionCursor ) { version = ( *versionCursor )->getString(); }
+  }
+
+  // Read `meta.description` field
+  std::optional<std::string> description;
+  {
+    auto descriptionCursor
+      = cursor->findAlongAttrPath( { state->sMeta, state->sDescription } );
+
+    if ( descriptionCursor )
+      {
+        description = ( *descriptionCursor )->getString();
+      }
+  }
+
+  std::optional<std::string> license;
+  {
+    // todo
+  }
+
+  std::optional<bool> broken;
+  {
+    auto brokenCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "broken" ) );
+
+    if ( brokenCursor ) { broken = ( *brokenCursor )->getBool(); }
+  }
+
+  std::optional<bool> unfree;
+  {
+    auto unfreeCursor
+      = cursor->findAlongAttrPath( nix::parseAttrPath( *state, "unfree" ) );
+
+    if ( unfreeCursor ) { unfree = ( *unfreeCursor )->getBool(); }
+  }
+
+
+  LockedInstallable lockedInstallable = {
+    .lockedUrl        = lockedUrl,
+    .flakeDescription = flakeDescription,
+    .lockedAttrPath   = lockedAttrPath,
+    .derivation       = derivation,
+    .outputs          = outputs,
+    .outputsToInstall = outputsToInstall,
+    .packageSystem    = systemAttribute,
+    .lockedSystem     = system,
+    .name             = name,
+    .pname            = pname,
+    .version          = version,
+    .description      = description,
+    .license          = license,
+    .broken           = broken,
+    .unfree           = unfree,
+  };
+
+  return lockedInstallable;
+}
+
+
+void
+to_json( nlohmann::json & jto, const LockedInstallable & from )
```

Comment:
I guess yeah, i didnt want to spend too much time figuring out renaming with these macros and at least at the moment defining `{to,from}_json` seems more consistent

Reply:
I think the only reason we needed to have `{to,from}_json` was because we parsed the manifest in C++ so errors were user facing. This is an internal only API so I don't think we need the same level of control. That said if you think this is easier than the macro, I'm not opposed to leaving it, I just don't think we *need* to do it for consistency for things we're about to drop.

Reply:
I'm going to need to add a `from_json` in my PR - do you care if I change over to using macros?

Reply:
no sure if you figure out the renaming (or we do away with kebab-case)

Reply:
Ugh maybe I'll just do `from_json` because kebab case